### PR TITLE
chore: deprecate an option `platform-user` in favour of whitelist

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Hard-coded restrictions for platform-user are super-seeded by whitelist feature.

## What was done?
Before actually removing feature, let's make it deprecated for now
Split from https://github.com/dashpay/dash/pull/6105/
6105 - to close and re-open for next major release.

It deprecates old command line argument `-platform-user` by renaming to `-deprecated-platform-user`

## How Has This Been Tested?
See new 2 functional tests: `rpc_deprecated_platform_filter.py` and `rpc_external_queue.py` which are split from `rpc_platform_filter.py`

## Breaking Changes
Command line argument `-platform-user` is renamed to `-deprecated-platform-user`

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/init.cpp [hunk 1]
```diff
@@ -734,7 +734,7 @@ void SetupServerArgs(NodeContext& node)
...
     argsman.AddArg("-masternodeblsprivkey=<hex>", "Set the masternode BLS private key and enable the client to act as a masternode", ArgsManager::ALLOW_ANY | ArgsManager::SENSITIVE, OptionsCategory::MASTERNODE);
-    argsman.AddArg("-platform-user=<user>", "Set the username for the \"platform user\", a restricted user intended to be used by Dash Platform, to the specified username.", ArgsManager::ALLOW_ANY, OptionsCategory::MASTERNODE);
+    argsman.AddArg("-deprecated-platform-user=<user>", "Set the username for the \"platform user\", a restricted user intended to be used by Dash Platform, to the specified username.", ArgsManager::ALLOW_ANY, OptionsCategory::MASTERNODE);
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/init.cpp:737]: bump

**Review by UdjinM6 — APPROVED** [review]
utACK 668efc4f1a2d700b56334f2d439ea7135223caf5

**Review by PastaPastaPasta — COMMENTED** [review]
LGTM; gonna wait to review until 21.2 is branched off to approve / merge

**Review by PastaPastaPasta — APPROVED** [review]
utACK e2c66ae3e9eef0d59cbd2f509bad25292b9ee66f

**Pull request merged**